pkg/registry/consul_test.go

```diff
@@ -3,15 +3,605 @@
 package registry
 
 import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
 	"testing"
+	"time"
 
+	"github.com/google/uuid"
+	"github.com/hashicorp/consul/api"
 	"github.com/stretchr/testify/assert"
 )
 
 func TestCreateConsulClient(t *testing.T) {
 	config := ConsulConfig{}
 	c1, err := NewConsulClient(&config)
-	t.Log(err)
 	assert.NotNil(t, c1)
 	assert.Nil(t, err)
 }
+
+func TestRegisterSuccess(t *testing.T) {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/v1/session/create", func(res http.ResponseWriter, req *http.Request) {
```

Comment:
nit: Seeing a lot of duplicate code. Maybe init two common HTTP servers (one for success and one for failure) with all routes and re-use. Use [TestMain](https://golang.org/pkg/testing/#hdr-Main)?

Reply:
issue with TestMain is that that is not per package, had explored using init() in file but there is no reverse for init 
Per test made sense as generally we define mock data as part of tests
if we create S and F as 2 separate things failures can have multiple types, creating combinations of failures will eventually mean defining those many combinations :(